File: src/test/java/com/pduda/field/ShoppingCartTest.java
package com.pduda.field;

import org.junit.Assert;
import org.junit.Test;

public class ShoppingCartTest {


    @Test
    public void singleItem_numberOfProductsInTheCart() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart(10);

        Assert.assertEquals(1, shoppingCart.numberOfProducts());
    }

    @Test
    public void singleItem_totalPrice() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart(10);

        Assert.assertEquals(10, shoppingCart.calculateTotalPrice());
    }

    @Test
    public void singleItem_hasDiscountIfContainsAProductWorthAtLeast100() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart(100);

        Assert.assertTrue(shoppingCart.hasDiscount());
    }

    @Test
    public void singleItem_doesNotHaveDiscountIfContainsAProductWorthLessThan100() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart(99);

        Assert.assertFalse(shoppingCart.hasDiscount());
    }

}

File: Cpp/Source/Field.Test/source/ShoppingCartTest.cpp

#include "Field/include/ShoppingCart.h"

#include <gtest/gtest.h>

TEST( singleItem_numberOfProductsInTheCart, ShoppingCartTest )
{
    ShoppingCart shoppingCart;
    shoppingCart.add(10);

    EXPECT_EQ(1, shoppingCart.numberOfProducts());
}

TEST( singleItem_totalPrice, ShoppingCartTest )
{
    ShoppingCart shoppingCart;
    shoppingCart.add(10);

    EXPECT_EQ(10, shoppingCart.calculateTotalPrice());
}

TEST( singleItem_hasDiscountIfContainsAtLeastOneProductWorthAtLeast100, ShoppingCartTest )
{
    ShoppingCart shoppingCart;
    shoppingCart.add(100);

    EXPECT_TRUE(shoppingCart.hasDiscount());
}

TEST( singleItem_doesNotHaveDiscountIfContainsNoProductsWorthAtLeast100, ShoppingCartTest )
{
    ShoppingCart shoppingCart;
    shoppingCart.add(99);

    EXPECT_FALSE(shoppingCart.hasDiscount());
}
